lib/Sema/TypeChecker.cpp

```diff
@@ -384,6 +384,10 @@ TypeCheckPrimaryFileRequest::evaluate(Evaluator &eval, SourceFile *SF) const {
       Ctx.evaluator,
       CheckInconsistentWeakLinkedImportsRequest{SF->getParentModule()}, {});
 
+  // Opt-in performance hint diagnostics for performance-critical code.
+  if (Ctx.LangOpts.hasFeature(Feature::SwiftPerformanceHints))
+    evaluateOrDefault(Ctx.evaluator, EmitPerformanceHints{SF}, {});
```

Comment:
FWIW, it's possible to check if specific warnings that are `DefaultIgnore` have been enabled. That would work here instead of having a separate experimental feature.

Reply:
Implemented this instead as a check for whether any of the relevant default-ignored diagnostics are enabled in the current compilation.